data/src/main/java/org/oppia/data/backends/gae/NetworkModule.kt

```diff
@@ -3,10 +3,8 @@ package org.oppia.data.backends.gae
 import dagger.Module
 import dagger.Provides
 import okhttp3.OkHttpClient
-import org.oppia.data.backends.gae.api.ClassroomService
-import org.oppia.data.backends.gae.api.TopicService
 import retrofit2.Retrofit
-import retrofit2.converter.moshi.MoshiConverterFactory
+import retrofit2.converter.gson.GsonConverterFactory
```

Comment:
Have you confirmed that Gson is a viable replacement to Moshi?

Reply:
I won't be able to fully confirm until I try it out on the real data module. What should I do in this case? I can't keep moshi and have the Bazel binary build. The final app library depends on NetworkModule - should I explore removing that dependency in some way?

Reply:
The only part of this module that depends on Moshi/GSON is the provider for Retrofit. You could abstract away Retrofit to avoid this module needing to depend on those libraries, similar to what you've done elsewhere.

Reply:
Okay - fixing this.

Reply:
So, I tried creating an interface but quickly realized that either way - moshi will be a necessary dependency for building app module. So, I thought that since data is not being used at the moment, and ApplicationComponent is the only file in app depending on NetworkModule, I would just remove NetworkModule from ApplicationComponent. If you agree with this solution, I could add a TODO to add NetworkModule back to ApplicationComponent once data is working?

Reply:
Can you confirm that the app builds and all tests work in both Bazel and Gradle with that module missing? If so, this sounds reasonable.

Reply:
Yes, can confirm. Also, created an issue and added a TODO.

Reply:
Thanks!